### PR TITLE
GH-16274: Fix plotting in explain: FigureCanvasAgg is non-interactive, and thus cannot be shown plt.show()

### PR DESCRIPTION
**Pull request by tomasfryda**
#16274

### CODE REVIEW

**Review by valenad1 — APPROVED** [review]
LGTM, thank you!

**Review by wendycwong — APPROVED** [review]

File: h2o-py/h2o/plot/_matplotlib.py [hunk 1]
```diff
@@ -3,7 +3,8 @@ def get_matplotlib_pyplot(server, raise_if_not_available=False):
     try:
         # noinspection PyUnresolvedReferences
         import matplotlib
-        matplotlib.use("Agg")
+        if server:
+            matplotlib.use("Agg")
```
Comment (wendycwong) [inline h2o-py/h2o/plot/_matplotlib.py:7]: @tomasfryda :  Can you add a comment here that if anything changes, please do an interactive test and outline what that interactive test should be?

**Review by wendycwong — CHANGES_REQUESTED** [review]

File: h2o-py/h2o/plot/_matplotlib.py (hunk 1 above)
Comment (tomasfryda) [reply h2o-py/h2o/plot/_matplotlib.py:8]: I added some general comment about testing explain module but I can't say what exactly should the person test. It depends on the changes. In this case any interactive test of plotting functionality would tell you that it's broken. But it's highly dependent on the change and in general changing things around matplotlib is not easy - we support a lot of versions and they behave differently so we should test for multiple versions of matplotlib interactively. And the test shouldn't be just seeing if there is an output - people use the plots to validate their models so we should make sure that the changes don't change the meaning of the plots or if it makes them useless due to some visual artifact, e.g., multiple colors replaced by one, in ice plot predicted value not plotted, wrong labels (i.e. labels don't correspond with the data) etc.

Unfortunately, it's a lot of work testing the plotting functionality but I haven't found a better way than checking everything manually.

**Review by wendycwong — APPROVED** [review]
Tomas:  I would like to see more description of what tests needed to be checked manually.  I will open up an new issue for you.  You can write up every tests needed in the issue and then come back and add the URL to your comments.  However, for the time being, let's get this fix in.

**Pull request merged**